### PR TITLE
[SIW-1820] Updated error handling in APIM policy

### PR DESCRIPTION
**Pull request by silvicir**
#### List of Changes

1. Updated `getCurrentWalletInstanceStatus` endpoint with fiscal code in header
2. Set `AppInsightsConnectionString` in key vault
3. Added trial system feature flag in APIM policy
4. Updated APIM policy with `403` returned status code when trial system returns an error

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by grausof** [comment]
is there a PR related for io-backend?

File: infra/resources/_modules/function_apps/locals.tf [hunk 1]
```diff
@@ -65,7 +65,11 @@ locals {
       WEBSITE_SWAP_WARMUP_PING_PATH     = "/api/v1/wallet/health"
       WEBSITE_SWAP_WARMUP_PING_STATUSES = "200"
       },
-      local.function_apps.common_app_settings
+      local.function_apps.common_app_settings,
+      {
+        for s in var.support_func.app_settings :
+        s.name => s.key_vault_secret_name != null ? "@Microsoft.KeyVault(VaultName=${var.project}-wallet-kv-01;SecretName=${s.key_vault_secret_name})" : s.value
+      }
```
Comment (grausof) [inline infra/resources/_modules/function_apps/locals.tf:72]: Why is this hardcoded?

**Comment by silvicir** [comment]
> is there a PR related for io-backend?

no

File: infra/resources/_modules/function_apps/locals.tf (hunk 1 above)
Comment (silvicir) [reply infra/resources/_modules/function_apps/locals.tf]: changed in f6dc59526dcd5a8b72dee0e5219ee081873f1401

**Review by grausof — APPROVED** [review]

**Pull request merged**